src/server/middleware/repoproxy/proxy.go

```diff
@@ -51,7 +51,7 @@ func handleBlob(w http.ResponseWriter, r *http.Request, next http.Handler) error
 	if err != nil {
 		return err
 	}
-	if !canProxy(p) || proxyCtl.UseLocalBlob(ctx, art) {
+	if proxyCtl.UseLocalBlob(ctx, art) || !canProxy(p) {
```

Comment:
What's the difference of the changing? Is there any special reason for that?

Reply:
Updated, UseLocalBlob will start HTTP request to check the status, it is should always after the canProxy check.